### PR TITLE
docker-setup interactive script 

### PR DESCRIPTION
**Pull request by guruswarupa**
# Pull Request

## Title
cleaner docker-setup for arch, debian and rpm based distros

## Type of Change
- closes #306  
- interactive script to install docker + docker-compose if users use yml files.
- tested on distrobox for each distro

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by guruswarupa** [comment]
@nnyyxxxx ig i've followed posix syntax this time have a check :)

**Comment by guruswarupa** [comment]
and thanks for helping me understand how to workaround certain basic github problems we face. got to learn a lot :)

**Comment by nnyyxxxx** [comment]
@guruswarupa IMO the other docker-setup PR is fine; after the final reviews are solved, it should be ready for merge.

I'd recommend closing this for now.

**Comment by guruswarupa** [comment]
> @guruswarupa IMO the other docker-setup PR is fine; after the final reviews are solved, it should be ready for merge.
> 
> I'd recommend closing this for now.

well yes, his doesn't include docker-compose which most them use when using docker. + its better to make it interactive than just installing it directly. also his code has too many systemctl commands which i believe the script already does that when i ran in debian. there was no need to start or enable the service.

File: src/commands/system-setup/6-docker-setup.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        [ "$INSTALL_DOCKER" -eq 1 ] && install_component "Docker" "docker" "curl -fsSL https://get.docker.com | sh"
+        [ "$INSTALL_COMPOSE" -eq 1 ] && install_component "Docker Compose" "docker compose" "$ESCALATION_TOOL $PACKAGER update && $ESCALATION_TOOL $PACKAGER install -y docker-compose-plugin"
+    elif [ "$PACKAGER" = "pacman" ]; then
+        [ "$INSTALL_DOCKER" -eq 1 ] && install_component "Docker" "docker" "$ESCALATION_TOOL $PACKAGER -S --noconfirm docker && $ESCALATION_TOOL systemctl enable docker && $ESCALATION_TOOL systemctl start docker"
+        [ "$INSTALL_COMPOSE" -eq 1 ] && install_component "Docker Compose" "docker-compose" "$ESCALATION_TOOL $PACKAGER -S --noconfirm docker-compose"
+    elif [ "$PACKAGER" = "yum" ]; then
+        [ "$INSTALL_DOCKER" -eq 1 ] && install_component "Docker" "docker" "curl -fsSL https://get.docker.com | sh"
+        [ "$INSTALL_COMPOSE" -eq 1 ] && install_component "Docker Compose" "docker compose" "$ESCALATION_TOOL $PACKAGER update && $ESCALATION_TOOL $PACKAGER install -y docker-compose-plugin"
+    fi
+}
```
Comment (nnyyxxxx) [inline src/commands/system-setup/6-docker-setup.sh:46]:
```bash
install_docker() {
    clear
    choose_installation
    if [ "$PACKAGER" = "apt-get" ]; then
        [ "$INSTALL_DOCKER" -eq 1 ] && install_component "Docker" "docker" "curl -fsSL https://get.docker.com | sh"
        [ "$INSTALL_COMPOSE" -eq 1 ] && install_component "Docker Compose" "docker compose" "$ESCALATION_TOOL $PACKAGER update && $ESCALATION_TOOL $PACKAGER install -y docker-compose-plugin"
    elif [ "$PACKAGER" = "pacman" ]; then
        [ "$INSTALL_DOCKER" -eq 1 ] && install_component "Docker" "docker" "$ESCALATION_TOOL $PACKAGER -S --noconfirm docker && $ESCALATION_TOOL systemctl enable docker && $ESCALATION_TOOL systemctl start docker"
        [ "$INSTALL_COMPOSE" -eq 1 ] && install_component "Docker Compose" "docker-compose" "$ESCALATION_TOOL $PACKAGER -S --noconfirm docker-compose"
    elif [ "$PACKAGER" = "yum" ]; then
        [ "$INSTALL_DOCKER" -eq 1 ] && install_component "Docker" "docker" "curl -fsSL https://get.docker.com | sh"
        [ "$INSTALL_COMPOSE" -eq 1 ] && install_component "Docker Compose" "docker compose" "$ESCALATION_TOOL $PACKAGER update && $ESCALATION_TOOL $PACKAGER install -y docker-compose-plugin"
    fi
}
```
This is unreadable.

Comment (guruswarupa) [reply src/commands/system-setup/6-docker-setup.sh:46]: The install component accepts package name to display the user while installing package, package itself and the commands to install, Which is given as arguments covered with three " ". What's there in it to not understand? Used && to combine the commands.

Comment (nnyyxxxx) [reply src/commands/system-setup/6-docker-setup.sh:46]: Here's a codeblock taken from one of the other scripts as an example
```bash
install_theme_tools() {
    printf "${YELLOW}Installing theme tools (qt6ct and kvantum)...${RC}\n"
    case $PACKAGER in
        apt-get)
            $ESCALATION_TOOL apt-get update
            $ESCALATION_TOOL apt-get install -y qt6ct kvantum
            ;;
        zypper)
            $ESCALATION_TOOL zypper refresh
            $ESCALATION_TOOL zypper --non-interactive install qt6ct kvantum
            ;;
        dnf)
            $ESCALATION_TOOL dnf update
            $ESCALATION_TOOL dnf install -y qt6ct kvantum
            ;;
        pacman)
            $ESCALATION_TOOL pacman -S --needed --noconfirm qt6ct kvantum
            ;;
        *)
            printf "${RED}Unsupported package manager. Please install qt6ct and kvantum manually.${RC}\n"
            exit 1
            ;;
    esac
}
```

File: src/commands/system-setup/6-docker-setup.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    
+    if ! command_exists "$command"; then
+        printf "%b\n" "${YELLOW}Installing $name...${RC}"
+        eval "$install_cmd"
+        printf "%b\n" "${GREEN}$name installed successfully.${RC}"
+    fi
+}
+
+install_docker() {
+    clear
```
Comment (nnyyxxxx) [inline src/commands/system-setup/6-docker-setup.sh:34]: ``clear``
No need for this.

Comment (guruswarupa) [reply src/commands/system-setup/6-docker-setup.sh:34]: @nnyyxxxx so you want the output the checkenv and checkEscalationTool to be displayed every single time user runs the script?  
image

Comment (nnyyxxxx) [reply src/commands/system-setup/6-docker-setup.sh:34]: @guruswarupa I don't see why not; that's how It's like in the other scripts.

Comment (guruswarupa) [reply src/commands/system-setup/6-docker-setup.sh]: those will later be changed anyways :)

Comment (guruswarupa) [reply src/commands/system-setup/6-docker-setup.sh]: > those will later be changed anyways :)

i get the idea on why to display those. to basically let the user who runs script know whats happenning, but when its interactive script, it doesn't make sense displaying it.

Comment (nnyyxxxx) [reply src/commands/system-setup/6-docker-setup.sh]: @guruswarupa I understand now, thanks for clarifying.

Comment (guruswarupa) [reply src/commands/system-setup/6-docker-setup.sh]: all my interactive scripts use clear method to do that :) like bluetooth control and wifi control

**Comment by ChrisTitusTech** [comment]
Sorry for the inconvenience. We had a massive restructure of the codebase to improve future development. Because of this can you update your PR to the new structure. Thank you for your assistance and contribution.

**Comment by guruswarupa** [comment]
@ChrisTitusTech  @nnyyxxxx i've fixed those conflicts. ready to be merged.

**Pull request merged**